assembl/static2/js/app/components/administration/discussion/timelineForm.jsx

```diff
@@ -0,0 +1,124 @@
+// @flow
+import React from 'react';
+import { I18n, Translate } from 'react-redux-i18n';
+import { OverlayTrigger, Row, Col } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import range from 'lodash/range';
+import { addPhaseTooltip } from '../../common/tooltips';
+import SectionTitle from '../sectionTitle';
+import PhaseForm from './phaseForm';
+import { createRandomId } from '../../../utils/globalFunctions';
+import { createPhase } from '../../../actions/adminActions/timeline';
+import PhaseTitleForm from './phaseTitleForm';
+
+type TimelineFormProps = {
+  editLocale: string,
+  phases: Array<string>,
+  handleCreatePhase: Function
+};
+
+type TimelineFormState = {
+  selectedPhaseId: string
+};
+
+export class DumbTimelineForm extends React.Component<TimelineFormProps, TimelineFormState> {
+  constructor(props: TimelineFormProps) {
+    super(props);
+    this.state = {
+      selectedPhaseId: props.phases ? props.phases[0] : ''
+    };
+  }
+
+  componentWillReceiveProps(nextProps: TimelineFormProps) {
```

Comment:
`componentWillReceiveProps` will be deprecated, please use `getDerivedStateFromProps`
It will not work for the phase creations (a few lines below) but I'm not sure you want to do this here